doc/move_group_interface/src/move_group_interface_tutorial.cpp

```diff
@@ -78,74 +78,38 @@ class MoveGroupInterfaceTutorial
     // Set the location where text is diplayed to above the robot
     text_pose_.translation().z() = 1.25;
   }
-  /**
-  * Publishes the given message above the robot
-  **/
-  void publishToRViz(std::string msg)
-  {
-    visual_tools_.deleteAllMarkers();
-    // To publish a text message inside Rviz
-    visual_tools_.publishText(text_pose_, msg, rvt::WHITE, rvt::XLARGE);
-    // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations
-    visual_tools_.trigger();
-    // This prompts the user before continuing.
-    visual_tools_.prompt("Press 'next' in the window to continue");
-  }
-
-  /**
-  * Publishes the given message above the robot, and displays a path
-  **/
-  void publishToRViz(std::string msg, const moveit::planning_interface::MoveGroupInterface::Plan& plan)
-  {
-    // Remove old paths and text
-    visual_tools_.deleteAllMarkers();
-    // To publish a text message inside Rviz
-    visual_tools_.publishText(text_pose_, msg, rvt::WHITE, rvt::XLARGE);
-    // To publish a planed path
-    visual_tools_.publishTrajectoryLine(plan.trajectory_, joint_model_group_);
-    // Batch publishing is used to reduce the number of messages being sent to RViz for large visualizations
-    visual_tools_.trigger();
-    // This prompts the user before continuing.
-    visual_tools_.prompt("Press 'next' in the window to continue");
-  }
 
   /**
   * Publishes the given message above the robot, displays a path, and points
   **/
-  void publishToRViz(std::string msg, const moveit::planning_interface::MoveGroupInterface::Plan& plan, std::vector<geometry_msgs::Pose> points)
+  void publishToRViz(std::string msg, const moveit::planning_interface::MoveGroupInterface::Plan* plan=nullptr, const std::vector<geometry_msgs::Pose>* points=nullptr, bool blocking=true, bool clear_prev=true)
```

Comment:
In C++ we don't use raw pointers unless we have to. If you need a pointer you should use a smart pointer like `std::shared_ptr` let me know if you need examples to work from

Reply:
Also, if you're only using `plan->trajectory_` then pass that in directly rather than passing in plan.


eg 
```
void publishToRViz(std::string msg, const moveit_msgs::RobotTrajectory& plan = moveit_msgs::RobotTrajectory())
{
  ...
  if (plan.joint_trajectory.points.size())
  {
    ...
  }
}
```

Reply:
Rather than use a pointer to a std::vector, you should make the default a std::vector<geometry_msgs::Pose> of length 0

Reply:
Done